src/Tizen.NUI/src/internal/Common/WeakEvent.cs

```diff
@@ -106,13 +124,46 @@ public WeakHandler(U handler)
                 methodInfo = d.Method;
             }
 
+            public bool Equals(object? otherTarget, MethodInfo otherMethodInfo)
+            {
+                return otherTarget == weakReference?.Target && otherMethodInfo.Equals(methodInfo);
+            }
+
             public bool Equals(U handler)
             {
                 Delegate other = (Delegate)(object)handler;
-                return other != null && other.Target == weakReference?.Target && other.Method.Equals(methodInfo);
+                var rooting = other?.Target;
+                return other != null && Equals(rooting, other.Method);
```

Comment:
```suggestion
                return other != null && (rooting == null ? EqualsByWeakReference(null, other.Method) : EqualsByTargetObject(rooting, other.Method));
```

We need to consider `weakHandleReference` is null cases.